source/game/field/CollisionDirector.hh

```diff
@@ -22,10 +22,41 @@ class CollisionDirector : EGG::Disposer {
     friend class Host::Context;
 
 public:
+    enum class eCollisionAttribute {
+        /*
+            |  0 - 4   |  5 - 7  | 8 | 9 | 10 |  11 - 12  |     13    |   14    |  15  |
+            | BaseType | Variant |   |   |    | Intensity | Trickable | Offroad | Soft |
+        */
+        Trickable = 13,
+        Offroad = 14,
+        Soft = 15
+    };
+    typedef EGG::TBitFlag<u16, eCollisionAttribute> CollisionAttribute;
+
     struct CollisionEntry {
         KCLTypeMask typeMask;
-        u16 attribute;
+        CollisionAttribute attribute;
         f32 dist;
+
+        // Credit: em-eight/mkw
+        /// Computes the "Base Type" portion of the KCL flags. It's the lower 5 bits of the flag.
+        u16 attributeBaseType() const {
+            return attribute & 0x1F;
+        }
+
+        /// Extracts the "Variant" portion of the KCL flag. It's the 3 bits before the "Bast Type".
+        u16 attributeVariant() const {
+            return (attribute >> 5) & 7;
+        }
+
+        u16 attributeIntensity() const {
+            return (attribute >> 11) & 3;
+        }
```

Comment:
We can infer that these methods pull from the attribute.
```suggestion
        /// Extracts the "Base Type" portion of the attribute.
        u16 baseType() const {
            return attribute & 0x1F;
        }

        /// Extracts the "Variant" portion of the attribute.
        u16 variant() const {
            return (attribute >> 5) & 7;
        }

        /// Extracts the "Intensity" portion of the attribute.
        u16 intensity() const {
            return (attribute >> 11) & 3;
        }
```
If you'd prefer to keep the bit comments explaining which specific bits it extracts, we could add `@brief` before the current comments and add the bits as `@details`.